Adjustments

Nightenom opened the pull request:
# Changes proposed in this pull request:
- fixes to latest blockout upgrade

Review please

Nightenom committed:
Update AbstractTextElement.java

Nightenom committed:
Annotane nullables in text element

Nightenom committed:
Clean up

Nightenom committed:
Update AbstractTextElement.java

ShadowProtocol committed:
Add get/setters and constructor overloads to ToggleButton

Raycoms commented on src/main/java/com/ldtteam/blockout/controls/AbstractTextElement.java:
```diff
@@ -396,20 +403,39 @@ public void setText(final List<IFormattableTextComponent> text)
+        setText(Collections.singletonList(text));
+    }
+
+    public void clearText()
```
just missing javadoc on those

Nightenom committed:
Jdoc, generic page shifting method in SwitchView

Nightenom committed:
Return page index

Raycoms commented:
:shipit:

Nightenom commented:
I might add more as I go through mcol code :D

Nightenom committed:
ITextHateSoMuchUselessAbstractionComponent

Nightenom committed:
Fix and make public ways for unlimited text folding

Nightenom committed:
make build system output sources for once

Nightenom committed:
Update build.gradle

Nightenom committed:
@ShadowProtocol take a look at this

You broke layout parsing cuz there is no layout constructor
This should basically fix it

Nightenom committed:
fix text builder colors, fix tooltip logic

Nightenom committed:
Another Shadow's blunder :open_mouth:

Nightenom committed:
Clear xml cache on resource reload

Nightenom committed:
Text, hover, scrolling group fixes

Nightenom committed:
Prepare for better text rendering

Nightenom committed:
Merge branch 'version/1.16.3' into adjustments

Nightenom committed:
Cleaning

Raycoms commented on src/main/java/com/ldtteam/blockout/Loader.java:
```diff
@@ -275,4 +280,9 @@ public static void addToCache(ResourceLocation loc, PaneParams doc)
         parsedCache.put(loc, new Tuple<>(1, doc));
     }
+
+    public static void cleanXMLparseCache()
```
jdoc

Raycoms commented on src/main/java/com/ldtteam/blockout/controls/AbstractTextElement.java:
```diff
@@ -268,6 +294,8 @@ else if (textAlignment.isHorizontalCentered())
             lineShift += mc.fontRenderer.FONT_HEIGHT + textLinespace;
         }
 
+        // TODO: forge disable filtering
```
todo for the future?

Nightenom replied:
todo for this <https://github.com/MinecraftForge/MinecraftForge/pull/7645>

Raycoms replied:
kk

Raycoms commented on src/main/java/com/ldtteam/blockout/controls/ToggleButton.java:
```diff
@@ -44,8 +76,32 @@ public ToggleButton(final PaneParams params)
+        return raw ? rawStates : states.stream().map(IFormattableTextComponent::getString).collect(Collectors.toList());
     }
 
+    public List<IFormattableTextComponent> getStates()
```
jdoc

ShadowProtocol replied:
We haven't been adding jdoc for pure getters like this one. It wouldn't really clarify anything further. Between all the getters around that's hundreds of lines of code we don't really need. There was a brief discussion on this in the original PR. This is what I said then:
> I just imagine it as if it is in an auto-genned doc page. would someone understand that this sets the text alignment based off of the method title alone? If the answer is yes then it doesn't need a jdoc IMO. You just need to be careful that you aren't doing other things in the setter that won't be described by the title alone. 
But if you're just going to chuck ``/** Sets the text alignment */`` above it then that is not worth anyone's energy scrolling past it.

But if you insist then I will.

Raycoms commented:
Looks fine for me, only missing some javadoc here and there :shipit: 

@OrionDevelopment @ShadowProtocol

ShadowProtocol committed:
jdoc

Raycoms committed:
don't replace different air with eachother

Nightenom committed:
Last fix and adjustment

Nightenom committed:
Merge branch 'version/1.16.3' into adjustments

Nightenom committed:
Merge branch 'version/1.16.3' into adjustments
